ui/packages/shared/functions/src/time.ts

```diff
@@ -11,7 +11,109 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+import format from 'date-fns/format';
 import intervalToDuration from 'date-fns/intervalToDuration';
+import {Duration} from '@parca/client';
+
+export interface TimeObject {
+  nanos?: number;
+  micros?: number;
+  milliseconds?: number;
+  seconds?: number;
+  minutes?: number;
+  hours?: number;
+  days?: number;
+  weeks?: number;
+  years?: number;
+}
+
+export const TimeUnits = {
+  Nanos: 'nanos',
+  Micros: 'micros',
+  Milliseconds: 'milliseconds',
+  Seconds: 'seconds',
+  Minutes: 'minutes',
+  Hours: 'hours',
+  Days: 'days',
+  Weeks: 'weeks',
+  Years: 'years',
+} as const;
+
+export type TimeUnit = typeof TimeUnits[keyof typeof TimeUnits];
+
+export const unitsInTime = {
+  [TimeUnits.Nanos]: {multiplier: 1, symbol: 'ns'},
+  [TimeUnits.Micros]: {multiplier: 1e3, symbol: 'µs'},
+  [TimeUnits.Milliseconds]: {multiplier: 1e6, symbol: 'ms'},
+  [TimeUnits.Seconds]: {multiplier: 1e9, symbol: 's'},
+  [TimeUnits.Minutes]: {multiplier: 6 * 1e10, symbol: 'm'},
+  [TimeUnits.Hours]: {multiplier: 60 * 60 * 1e9, symbol: 'h'},
+  [TimeUnits.Days]: {multiplier: 60 * 60 * 24 * 1e9, symbol: 'd'},
+  [TimeUnits.Weeks]: {multiplier: 60 * 60 * 24 * 7 * 1e9, symbol: 'w'},
+  [TimeUnits.Years]: {multiplier: 60 * 60 * 24 * 365 * 1e9, symbol: 'y'},
```

Comment:
Not sure if worth adding as a comment but wanted it to mention once. The Go team, for example, decided against having weeks and years as constants. Especially the year one is wrong once every 4 years. :grimacing: It works well enough for our use case and Prometheus does the same, just be aware of the fact.

Reply:
ok, that's good to know! You're right with leap years, it could all be off. Maybe I'll just add a TODO in the code while I'm fixing whats blocking the build so we can flag this and return to it down the road